site/docs/patterns/wizard.mdx

```diff
@@ -0,0 +1,278 @@
+
+You can use a dialog component as a structure base for a modal wizard with a stepper when you need to insert the wizard within another task or flow. The header in the dialog should include the wizard title and stepper. 
+
+For shorter tasks that temporarily become the user’s main focus, a modal experience is preferable, as it prevents interaction with the rest of the UI and helps focus the user’s attention on the specific task or information at hand.
+
+<Diagram
+  src="/img/patterns/wizard/wizard-pattern-modal.png"
+  srcDark="/img/patterns/wizard/wizard-pattern-modal-dark.png"
+  alt="Modal wizard example with stepper and dialog container"
+  border
+  background="secondary"
+/>
+
+<GuidanceCallout type="positive" customPillText="Best practices">
+Do not include a close button, as it can be ambiguous whether the progress or the entered information will be saved or lost. Use a cancel button instead to clearly indicate the user’s intent to exit the flow, ensuring a conscious, deliberate action to stop the process.
+</GuidanceCallout>
+
+## Optional step
+
+When a step is entirely optional and contains no required fields, you can indicate this beneath the step title (in stepper). We also recommend to use the “All fields are optional” text in the description under the step title.
+
+For optional steps, keep the next button enabled so users can easily skip the step if they wish.
+
+<Diagram
+  src="/img/patterns/wizard/wizard-pattern-optional-step.png"
+  srcDark="/img/patterns/wizard/wizard-pattern-optional-step-dark.png"
+  alt="Wizard step with header description indicating all fields are optional"
+  border
+  background="secondary"
+  caption="“All fields are optional” text in the description under the step title"
+/>
+
+<GuidanceCallout type="positive" customPillText="Best practices">
+  Indicate the minority: When a step contains both required and optional fields, label only the few optional or required fields as needed.
+</GuidanceCallout>
+
+## Validation state
+
+Wizard can show validation feedback. Feedback can occur either in real time (dynamically) as the user interacts with the form or after submission (when clicking the final action button, e.g., Submit), depending on the validation approach implemented.
+
+When validation feedback is triggered:
+- The stepper updates to reflect the validation state.
+- All action buttons remain enabled throughout the flow for accessibility and discoverability.
+- You should give the user options to resolve issues before they proceed. 
+- Users can still go back or cancel the flow if needed.
+
+<GuidanceCallout type="positive" customPillText="Best practices">
+  When navigating back through previous steps, the current step is always updated to show ‘Active’ and the steps ahead of the users current step should always be shown as ‘Pending’ (this includes any steps which were showing validation errors, since these may be corrected, or become unnecessary as the user progresses through the wizard).
```

Comment:
```suggestion
  When navigating back through previous steps, the current step is always updated to show ‘Active’ and the steps ahead of the user's current step should always be shown as ‘Pending’ (this includes any steps which were showing validation errors, since these may be corrected, or become unnecessary as the user progresses through the wizard).
```